packages/react-client/src/hooks/peerStatus.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import type { PeerStatus } from "../state.types";
+import type { FishjamClient } from "@fishjam-cloud/ts-client";
+import type { PeerMetadata, TrackMetadata } from "../types";
+
+export const usePeerStatus = (client: FishjamClient<PeerMetadata, TrackMetadata>) => {
+  const [peerStatus, setPeerStatus] = useState<PeerStatus>(null);
+
+  useEffect(() => {
+    const setAuthenticated = () => {
+      setPeerStatus("authenticated");
+    };
+    const setError = () => {
+      setPeerStatus("error");
+    };
+    const setJoined = () => {
+      setPeerStatus("joined");
+    };
+    const setDisconnected = () => {
+      setPeerStatus(null);
+    };
+    const setConnected = () => {
+      setPeerStatus("connected");
+    };
```

Comment:
**Discussion idea**: Based on my experience, `joined` and `connected` are very confusing. I can't count how many times I made a mistake and considered the state `connected` as final and correct. The correct state is `joined`. I suggest in the future to replace `joined` with `connected`, but I have no idea what to call the old `connected`

Reply:
Makes sense, we could have us another "proposal" ticket.

Reply:
https://linear.app/swmansion/issue/FCE-458/rename-client-statuses-joined-connected